runtime/lua/vscode-neovim/api.lua

```diff
@@ -1,133 +1,302 @@
+  elseif value_is_table and not name_is_table then
+    error([[The "value" is a table, but the "name" is not]])
   end
+  assert(
+    target == nil or target == "global" or target == "workspace",
+    [["target" can only be nil or one from "global" and "workspace"]]
+  )
+  return M.call("update_config", { args = { name, value, target } })
 end
 
----Handle document changes
----@param bufnr number
----@param changes (string | integer)[][]
----@return number: changed tick of the buffer
-function M.handle_changes(bufnr, changes)
-  -- Save and restore local marks
-  -- Code modified from https://github.com/neovim/neovim/pull/14630
-  local marks = {}
-  for _, m in pairs(fn.getmarklist(bufnr or api.nvim_get_current_buf())) do
-    if m.mark:match("^'[a-z]$") then
-      marks[m.mark:sub(2, 2)] = { m.pos[2], m.pos[3] - 1 } -- api-indexed
+---------------------------
+------ Notifications ------
+---------------------------
+
+--- Display a notification to the user.
+---
+---@param msg string Content of the notification to show to the user.
+---@param level integer|nil One of the values from |vim.log.levels|.
+---@param opts table|nil Optional parameters. Unused by default.
+---@diagnostic disable-next-line: unused-local
+function M.notify(msg, level, opts)
+  local level_name
+  local levels = vim.log.levels
+  level = level or levels.INFO
+
+  -- legacy
+  if type(level) == "string" then
+    if level == "error" then
+      level = levels.ERROR
+    elseif level == "warn" then
+      level = levels.WARN
+    else
+      level = levels.INFO
     end
   end
 
-  for _, change in ipairs(changes) do
-    api.nvim_buf_set_text(bufnr, unpack(change))
+  if level >= levels.ERROR then
+    level_name = "error"
+  elseif level >= levels.WARN then
+    level_name = "warn"
+  else
+    level_name = "info"
   end
+  M.action("notify", { args = { msg, level_name } })
+end
 
-  local max = api.nvim_buf_line_count(bufnr)
-  -- no need to restore marks that still exist
-  for _, m in pairs(fn.getmarklist(bufnr or api.nvim_get_current_buf())) do
-    marks[m.mark:sub(2, 2)] = nil
-  end
-  -- restore marks
-  for mark, pos in pairs(marks) do
-    if pos then
-      -- make sure we don't go out of bounds
-      local line = (api.nvim_buf_get_lines(bufnr, pos[1] - 1, pos[1], false))[1] or ""
-      pos[1] = math.min(pos[1], max)
-      pos[2] = math.min(pos[2], #line)
-      api.nvim_buf_set_mark(bufnr or 0, mark, pos[1], pos[2], {})
+do
+  local notified = {}
+
+  --- Display a notification only one time.
+  ---
+  --- Like |vscode.notify()|, but subsequent calls with the same message will not
+  --- display a notification.
+  ---
+  ---@param msg string Content of the notification to show to the user.
+  ---@param level integer|nil One of the values from |vim.log.levels|.
+  ---@param opts table|nil Optional parameters. Unused by default.
+  ---@return boolean true if message was displayed, else false
+  function M.notify_once(msg, level, opts)
```

Comment:
What is the expected behavior with `notify_once("one"), notify_once("two"), notify_once("one")`? For it to ignore all duplicate messages, or only ones that are in a row?

Reply:
`vim.notify_once`

Reply:
Not sure you understood - what is the default behavior of vim.notify_once?

Reply:
https://github.com/neovim/neovim/blob/0892c1cd94672e0dffe3b97fc4a85ea517b05598/runtime/lua/vim/_editor.lua#L620

Reply:
Alright. If the neovim implementation calls vim.notify, we don't need to override it right? Only override vim.notify.

Reply:
Oh, right. I didn’t realize that.